langfuse/_task_manager/media_manager.py

```diff
@@ -0,0 +1,245 @@
+import logging
+from queue import Empty
+from typing import Any, Callable, Optional, TypeVar
+from typing_extensions import ParamSpec
+
+import time
+import requests
+import backoff
+
+from langfuse.api import GetMediaUploadUrlRequest, PatchMediaBody
+from langfuse.api.core import ApiError
+from langfuse.api.client import FernLangfuse
+from langfuse.media import LangfuseMedia
+from langfuse.utils import _get_timestamp
+
+from .media_upload_queue import MediaUploadQueue, UploadMediaJob
+
+
+T = TypeVar("T")
+P = ParamSpec("P")
+
+
+class MediaManager:
+    _log = logging.getLogger(__name__)
+
+    def __init__(
+        self,
+        *,
+        api_client: FernLangfuse,
+        media_upload_queue: MediaUploadQueue,
+        max_retries: Optional[int] = 3,
+    ):
+        self._api_client = api_client
+        self._queue = media_upload_queue
+        self._max_retries = max_retries
+
+    def process_next_media_upload(self):
+        try:
+            upload_job = self._queue.get(block=True, timeout=1)
+            self._log.debug(f"Processing upload for {upload_job['media_id']}")
+            self._process_upload_media_job(data=upload_job)
+
+            self._queue.task_done()
+        except Empty:
+            self._log.debug("Media upload queue is empty")
```

Comment:
LOL. this will be super noisy if debug logs enabled :D

Reply:
The queue job retrieval is with block=True and timeout of 1 second, so only one debug log per second :)